### ISSUE TITLE
CustomResources doesn't work if you implement using APIService with aggregated api-server

### ISSUE THREAD

**Issue by raushan2016** [Azure/vscode-kubernetes-tools#643]
There are 2 ways to build CRDs.

- Using Controllers and CRD yaml files. In this case you create object of type `apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition`
- Using APIService and writing your own api-server and wiring it up with aggregated api-server.
`apiVersion: apiregistration.k8s.io/v1
kind: APIService`

Currently extension works fine if you go with option1, but it doesn't work with option2.
If you go with option2, you don't see anything under "Custom Resources". My assumption is it does `kubectl get crds`.

But at the end both of path goes to `kubectl get abc.foobar.k8s.io` to get the individual crds.

Adding the support to discover the crds via both channel will make this extension more helpful.

**Comment by itowlson** [comment]
You are right, the `Custom Resources` folder does do `kubectl get crd`.  Is there a standard way to discover CRDs in the aggregated api-server design?  If not, we can work with you to build something custom that can hook up your particular CRDs.

**Comment by liggitt** [comment]
> Is there a standard way to discover CRDs in the aggregated api-server design?

aggregated resources are not CRDs. using the `/apis` endpoint to discover served groups/versions, then requesting `/apis/<group>/<version>` to discover served resources for that group/version is how clients are expected to discover available resources (regardless of whether they are built-in, crds, or aggregated)

**Comment by itowlson** [comment]
Okay, that makes sense.  We currently surface only specific resource kinds, so I guess we would need to request `/apis` and sort them into:

* Known resources, which have defined locations in the tree
* Custom resources (the `CustomResourceDefinition` type and any resources defined by an instance of `CustomResourceDefinition`), which go in the Custom Resources folder
* Everything else (which might be aggregated resources or stuff we don't currently provide a home for)

Then we could put the 'everything else' types under a folder such as 'Other Resources.'

Or we could have an All Resources folder and _everything_ goes in there, even things that also get displayed elsewhere in the tree.  That might be easier.

Would one of these display options work for you folks?  Do you have a preference?
